src/development/metrics.md

```diff
@@ -0,0 +1,153 @@
+
+```python
+class SuffixStateDiffs(Container):
+    updated_suffixes: bytes
+    updated_currents: List[Bytes32]
+    updated_news: List[Bytes32]
+
+    inserted_suffixes: bytes
+    inserted_news: List[Bytes32]
+
+    read_suffixes: bytes
+    read_currents: List[Bytes32]
+
+    missing_suffixes: bytes
+```
+Type 2 serialization container
+
+ * A final variation, where update, reads, inserts and proof-of-absences are grouped, along with their suffixes, into their own lists. This is referred to as **type 3**.
+
+```python
+class UpdateDiff(Container):
+    suffix: byte
+    current: Bytes32
+    new: Bytes32
+
+class ReadDiff(Container):
+    suffix: byte
+    current: Bytes32
+
+class InsertDiff(Container):
+    suffix: byte
+    new: Bytes32
+
+class MissingDiff(Container):
+    suffix byte
+
+class StemStateDiff(Container):
+    stem: Bytes31
+
+    updates: List[UpdateDiff]
+    reads: List[ReadStateDiff]
+    insert: List[InsertStateDiff]
+    # TODO: check if this encodes as well as bytes
+    missing: List[MissingStateDiff]
+```
+Type 3 serialization container
+
+Here are our findings after replaying data for 200k blocks:
+
+![image.png](./assets/total.png)
+
+One can see that the spec is much less efficient than the other types. This is due to the inefficiency of storing `Optional[T]`. As a result, we didn’t include this method in the rest of the analysis.
+
+When only considering the post-transition witnesses, this is what is found:
+
+![image.png](./assets/compare_post_transition.png)
+
+The maximum and minimum sizes for each type, are summarized in the following table:
+
+|Name|Min (KB)|Max (KB)|
+|-|-|-|
+|type 1|425|1285|
+|type 2|383|928|
+|type3|380|928|
+
+While type 2 and type 3 are quite close, one can see that type 3 has better average and minimum sizes.
+
+#### Witness size breakdown (type 3)
+
+Replaying past blocks, we generated the witnesses and could provide the following size breakdown for type 3 witnesses. This is averaged over ~134k blocks, as the conversion blocks were skipped.
+
+|Component|avg size %|min size %|max size %|
+|-|-|-|-|
+|keys|13|9|49|
+|pre-values|78|7|85|
+|post-values|0.1|0|47|
+|verkle proof|12|8|77|
+
+#### Takeaways
+
+The key takeaways are:
+
+ * Type 3 witnesses offer a great compression
+ * post-values are of dubious use, and can not be verified without block execution. Although they don’t take much space on average, we suggest dropping them from the witness.
+
+### Database
+
+Looking at a fully converted database (conversion + ~134k blocks replayed), we found that the leaf depths could be broken into :
+
+|Depth|Count|Percentage|
+|-|-|-|
+|4|745822024|80.6|
+|5|178484052|19.3|
+|6|775718|0.1|
+|7|2880|0|
+|8|10|0|
+
+### Gas usage
+
+## Binary-specific measurements
+
+## State expiry-specific measurements
```

Comment:
I would suggest removing sections without information since they're confusing for readers.

If this is set as a reminder, we should create these as issues in our internal tracking for this project.